### PR TITLE
Implemented MSAL Angular

### PR DESCRIPTION
**Pull request by Szsaunders**

### CODE REVIEW

**Comment by Szsaunders** [comment]
Environment variables are set up, but it won't build in production mode if I replace the B2C config info in AppModule with it (though it _will_ in dev mode), so that's on hold for now. 

Duplication issues are just because of copying over environment variables. Code coverage...is admittedly not great.

**Comment by Szsaunders** [comment]
Turns out the issue was that I was giving MsalModule a function, and you can't do that with module declarations?

File: src/environments/environment.ts [hunk 1]
```diff
@@ -6,35 +6,53 @@ import { SelectionEnvironment } from '@revaturexyz/selection';
...
+}
+
+
+export const AuthEnvironmentImpl: AuthEnvironment = {
+  clientID: 'd2bba178-c1a8-4257-95c2-9452123fcdc1',
+  authority: 'https://login.microsoftonline.com/te/revaturehousingtemp.onmicrosoft.com/B2C_1_signupsignin1',
+  validateAuthority: true,
+  cacheLocation: 'localStorage',
+  postLogoutRedirectUri: 'https://revaturexyz-housing-dev.southcentralus.cloudapp.azure.com'
+};
```
Comment (escalonn) [inline src/environments/environment.ts]: this does not really match how we are doing the rest; ___Impl is a class extending an abstract class, not an instance. iirc we are following [this approach](https://github.com/nrwl/nx/issues/208#issuecomment-475911897) here. i will let it through but you may need to rewire it in order for the profile module to truly receive that info, maybe.

i would think you could just add an auth environment field to profile environment, and pass it in alongside the rest of profile env values

**Comment by escalonn** [comment]
also i don't know why you added rxjs-compat package; there's never a reason to use that package in new code

**Pull request merged**